Assets/Scripts/Control&Input/ItemSelectSlot.cs

```diff
@@ -52,55 +60,80 @@ public void SetItems(List<Item> items, Item initialItem)
         {
             items.Insert(0, null);
         }
-
-        // After lots of tweaking, this formula seems to translate size correctly
-        var transformedSpacing = spacing * (itemModelScale / spacing) / 2f;
-
+        int itemCount = items.Count;
+        if (itemCount < 6)
+        {
+            for (int i = 0; i < 6 - itemCount; i++)
+            {
+                items.Insert(itemCount + i, items[Mod((itemCount + i), itemCount)]);
+            }
+        }
 
         // Instantiate all items at offset spacing down from holder
         for (var i = 0; i < items.Count; i++)
         {
-            if (items[i] == null) continue;
-
+            // Item is probably an empty extension slot!
+            if (items[i] == null) 
+            {
+                var missing = Instantiate(missingObject, Vector3.zero, itemHolder.rotation, itemHolder);
+                missing.transform.localPosition = Vector3.zero;
+                augmentModels.Add((missing, missing.transform.localPosition));
+                continue;
+            }
             var rotation = Quaternion.Euler(new Vector3(0, 90, -20));
-
-            var instance = Instantiate(items[i].augment, itemHolder.position + Vector3.down * spacing * i, rotation, itemHolder);
+            var instance = Instantiate(items[i].augment, Vector3.zero, rotation, itemHolder);
             instance.transform.ScaleAndParent(itemModelScale, itemHolder);
-            instance.transform.position = itemHolder.position + Vector3.down * transformedSpacing * i;
+            instance.transform.localPosition = itemSlots[i].transform.position;
 
             Augment.DisableInstance(instance, type);
+            augmentModels.Add((instance, instance.transform.position));
         }
 
         selectedIndex = items.IndexOf(initialItem);
 
         // For extensions, we may not find any available item
         if (selectedIndex < 0) selectedIndex = 0;
 
-        ChangeItemDisplayed(selectedIndex, false);
+        StartCoroutine(WaitAndSetWeapons());
+    }
+
+    private IEnumerator WaitAndSetWeapons()
+    {
+        yield return new WaitForSeconds(0.1f);
+        ChangeItemDisplayed(selectedIndex);
     }
 
-    public void Select()
+    public void Select(out Item item)
```

Comment:
I'm curious why you chose to use an out parameter instead of changing the return type... And why couldn't you just use the SelectedItem property directly? There's no reason for avoiding so that I can discern

Reply:
It was to avoid convoluted nullable annotation in the return type. My original thought was that there might be cases where one isn't interested in the out variable either, but currently there is no example of that left

Reply:
But still, why not use the SelectedItem property?

Reply:
I'm going to change it to using that property instead, I just forgot to change it since I removed all places where the optional out was relevant 😎